hibernate-core/src/main/java/org/hibernate/persister/entity/UnionSubclassEntityPersister.java

```diff
@@ -125,19 +169,143 @@ public UnionSubclassEntityPersister(
+			deleteResultCheckStyles[j] = join.getCustomSQLDeleteCheckStyle() == null
+					? ExecuteUpdateResultCheckStyle.determineDefault( customSQLDelete[j], deleteCallable[j] )
+					: join.getCustomSQLDeleteCheckStyle();
+
+			Iterator iter = join.getKey().getColumnIterator();
+			keyColumnNames[j] = new String[join.getKey().getColumnSpan()];
+			int i = 0;
+			while ( iter.hasNext() ) {
+				Column col = (Column) iter.next();
+				keyColumnNames[j][i++] = col.getQuotedName( factory.getDialect() );
+			}
+		}
 
+		final boolean lazyAvailable = isInstrumented();
+
+		boolean hasDeferred = false;
+		ArrayList<String> subclassTableClosure = new ArrayList<String>();
+		ArrayList<String[]> joinKeyColumns = new ArrayList<String[]>();
+		ArrayList<Boolean> isConcretes = new ArrayList<Boolean>();
+		ArrayList<Boolean> isDeferreds = new ArrayList<Boolean>();
+		ArrayList<Boolean> isInverses = new ArrayList<Boolean>();
+		ArrayList<Boolean> isNullables = new ArrayList<Boolean>();
+		ArrayList<Boolean> isLazies = new ArrayList<Boolean>();
+		subclassTableClosure.add( tableNames[0] );
+		joinKeyColumns.add( getIdentifierColumnNames() );
+		isConcretes.add( Boolean.TRUE );
+		isDeferreds.add( Boolean.FALSE );
+		isInverses.add( Boolean.FALSE );
+		isNullables.add( Boolean.FALSE );
+		isLazies.add( Boolean.FALSE );
+		joinIter = persistentClass.getSubclassJoinClosureIterator();
+		while ( joinIter.hasNext() ) {
+			Join join = (Join) joinIter.next();
+			isConcretes.add( persistentClass.isClassOrSuperclassJoin( join ) );
+			isDeferreds.add( join.isSequentialSelect() );
+			isInverses.add( join.isInverse() );
+			isNullables.add(
+					join.isOptional() || creationContext.getSessionFactory()
+							.getSessionFactoryOptions()
+							.getJpaCompliance()
+							.isJpaCacheComplianceEnabled()
+			);
+			isLazies.add( lazyAvailable && join.isLazy() );
+			if ( join.isSequentialSelect() && !persistentClass.isClassOrSuperclassJoin( join ) ) {
+				hasDeferred = true;
+			}
+			subclassTableClosure.add( determineTableName( join.getTable(), jdbcEnvironment ) );
+			Iterator iter = join.getKey().getColumnIterator();
+			String[] keyCols = new String[join.getKey().getColumnSpan()];
+			int i = 0;
+			while ( iter.hasNext() ) {
+				Column col = (Column) iter.next();
+				keyCols[i++] = col.getQuotedName( factory.getDialect() );
+			}
+			joinKeyColumns.add( keyCols );
+		}
+
+		subclassTableSequentialSelect = ArrayHelper.toBooleanArray( isDeferreds );
+		subclassTableNameClosure = ArrayHelper.toStringArray( subclassTableClosure );
+		subclassTableIsLazyClosure = ArrayHelper.toBooleanArray( isLazies );
+		subclassTableKeyColumnClosure = ArrayHelper.to2DStringArray( joinKeyColumns );
+		isClassOrSuperclassTable = ArrayHelper.toBooleanArray( isConcretes );
+		isInverseSubclassTable = ArrayHelper.toBooleanArray( isInverses );
+		isNullableSubclassTable = ArrayHelper.toBooleanArray( isNullables );
+//		hasSequentialSelects = hasDeferred; TODO
```

Comment:
I didn't implement sequential selects here, but I'm also unsure how it should be implemented, and whether we even technically can support sequential selects for this type of persistence.